### PR TITLE
create cypress test for filter on users page

### PR DESCRIPTION
**Pull request by ShaiahWren**
This pr addresses issue https://issues.redhat.com/browse/AAH-785

Cypress testing for user filter, checking for exact and partial matches on each dropdown element: username, first name, last name, and email.

Functions should be for selectors uses multiple times throughout the test.

### CODE REVIEW

**Comment by himdel** [comment]
@ShaiahWren the `pr-checks` failure will go away after `npm install ; npm run prettier`, we updated the version in #351, so until you run npm install, it's trying to use the older rules.

**Comment by ShaiahWren** [comment]
@ZitaNemeckova @himdel Do you have a chance to review this one? :)

**Review by himdel — APPROVED** [review]

**Comment by ShaiahWren** [comment]
Pr closed. :)

File: test/cypress/integration/user_filter.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    cy.get('#email').type('new_user@example.com');
+    cy.get('#password').type('veryhardpassword');
+    cy.get('#password-confirm').type('veryhardpassword');
+    cy.get('.pf-c-button.pf-m-primary').contains('Save').click();
+  });
+
+  it('filters by username', () => {
+    usernamefilterInput().type('new_user');
+    search();
+    checkUserEntry();
```
Comment (himdel) [inline test/cypress/integration/user_filter.js:79]: Apologies, I missed this in the review...

Individual tests should not depend on each other .. `checkUserEntry` now tests for existence of the user created in the previous test.

Also the test is creating a user but not using a `before` block to clean up ;).

(Noticed while fixing (possibly unrelated) master cypress failures, fixing in https://github.com/ansible/ansible-hub-ui/pull/751)

**Pull request merged**